subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -508,7 +508,10 @@ static void mqtt_evt_handler(struct mqtt_client *const client,
 			LOG_ERR("Connection was rejected with return code %d",
 				mqtt_evt->param.connack.return_code);
 			LOG_WRN("Is the device certificate valid?");
```

Comment:
PS: Not a part of the changes in this PR. A connection fail here could be due to not just the device certificate? but also the private/public keypair. Maybe a more generic message such as "Is the device credentials valid" is better.

Reply:
The most common case in my testing has been certificate issues. Could add some more checking of the actual response code in CONNACK message in the future